packages/language-server/src/server.ts

```diff
@@ -112,11 +98,91 @@ export function startServer(options?: LSOptions): void {
     return result
   })
 
+  connection.onInitialized(() => {
+    if (hasConfigurationCapability) {
+      // Register for all configuration changes.
+      // eslint-disable-next-line @typescript-eslint/no-floating-promises
+      connection.client.register(
+        DidChangeConfigurationNotification.type,
+        undefined,
+      )
+    }
+  })
+
+  // The global settings, used when the `workspace/configuration` request is not supported by the client or is not set by the user.
+  // This does not apply to VSCode, as this client supports this setting.
+  const defaultSettings: LSPSettings = {
+    prismaFmtBinPath: defaultBinPath,
+  }
+  let globalSettings: LSPSettings = defaultSettings
+
+  // Cache the settings of all open documents
+  const documentSettings: Map<string, Thenable<LSPSettings>> = new Map<
+    string,
+    Thenable<LSPSettings>
+  >()
+
+  connection.onDidChangeConfiguration((change) => {
+    connection.console.info('Configuration changed.')
+    if (hasConfigurationCapability) {
+      // Reset all cached document settings
+      documentSettings.clear()
+    } else {
+      globalSettings = <LSPSettings>(
+        (change.settings.prismaLanguageServer || defaultSettings) // eslint-disable-line @typescript-eslint/no-unsafe-member-access
+      )
+    }
+    documents.all().forEach(async (d) => await installPrismaFmt(d.uri)) // eslint-disable-line @typescript-eslint/no-misused-promises
+
+    // Revalidate all open prisma schemas
+    documents.all().forEach(validateTextDocument) // eslint-disable-line @typescript-eslint/no-misused-promises
```

Comment:
Just a small point, we can also do this when a document is "opened". This is also okay because I don't expect too many Prisma schema files opened at once.

Reply:
Do you mean installing the binary and validating for all documents, not just the one being opened?

Reply:
I meant it for just the validation part. This is also good though 👍